src/app/users.component.html

```diff
@@ -2,45 +2,38 @@ <h2>Users</h2>
 <div class="toolbar">
   <mat-form-field class="search">
     <mat-icon matPrefix>search</mat-icon>
-    <input #in (input)="data.queryBy({search: in.value})" type="text" matInput placeholder="Search">
+    <label>
+      <input #input (input)="data.queryBy({search: input.value})" type="text" matInput placeholder="Search">
+    </label>
   </mat-form-field>
   <mat-form-field class="registration">
-    <input (dateChange)="data.queryBy({registration: $event.value})" matInput [matDatepicker]="picker" placeholder="Registration"/>
+    <label>
+      <input (dateChange)="data.queryBy({registration: $event.value})" matInput [matDatepicker]="picker" placeholder="Registration"/>
+    </label>
     <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
     <mat-datepicker #picker></mat-datepicker>
   </mat-form-field>
-  <mat-form-field class="property">
```

Comment:
I'm going back to work on that

Reply:
So here is the problem with that: https://github.com/angular/components/issues/10242

Reply:
Ah, that's too bad. Doesn't even work when it's bound via template? :thinking: 
Hm, so one of the following, right?
- one sorting input
- hack around the issue
- have two separate examples

Reply:
Yes that's the case, for sake of simplicity I'd suggest two different examples, because it will make it easier to read and understand.

Reply:
Yeah, let's go with that

Reply:
So I did split the example into two parts

Reply:
Thanks, this looks really good, like the tabs. Would you mind using `(matSortChange)` for the sort changes though? You could bind it to this method instead of having ngOnInit:
```typescript
sortBy({active, direction}: Sort): void {
  this.data.sortBy({
    property: active as keyof User,
    order: direction || 'asc'
  })
}
```
```html
(matSortChange)="sortBy($event)"
```

Otherwise you'd need to unsubscribe via ngOnDestroy somehow.

Reply:
Nice I did not think of the matSortChange. I'll force push the change now.